dev/implementations/storage_definitions.h

```diff
@@ -78,8 +78,9 @@ namespace sqlite_orm {
                             res = decltype(res)::new_columns_added_and_old_columns_removed;
                         }
                     } else if(schema_stat == sync_schema_result::dropped_and_recreated) {
-                        this->drop_table_internal(tImpl.table.name, db);
-                        this->create_table(db, tImpl.table.name, tImpl);
+                        this->backup_table(db, tImpl, {});

```

Comment:
this will break what `sync_schema` is designed for: schema synchronization. `sync_schema` must synchronize schema first, if data cannot be saved it will be deleted instead of operation abort. If we will abort schema synchronization in cases when data cannot be stored then user will be unable to sync schema in some circumstances. That is not right

Reply:
anyway you need to run `sync_schema` from within try/block cause it can throw just like any SQLite call. But it doesn't change my words

Reply:
> if data cannot be saved it will be deleted instead of operation abort
the operation abort will give the programmer a chance to see what is preventing the migration and take action so it can